DoSomething/northstar
# Adds PasswordResetType

#### What's this PR do?

This PR adds a `PasswordResetType` class to begin defining various types of password reset emails we'll send, such as the existing Forgot Password type and the upcoming Rock The Vote Activate Account type.

* It also renames the `ResetPassword` mail from #841 as `PasswordReset`, because  `PasswordResetType` felt more readable than `ResetPasswordType` ... and also because the end goal is to post to the `events/user-password-reset` Blink endpoint. I kept going back and forth on this, as Laravel uses both the `PasswordReset` and `ResetPassword` naming within existing classes/functions.

* Refactors the currently deprecated `POST /resets` endpoint:
    * Requires a `type` parameter, expecting a  valid `PasswordResetType` 
    * Send a PasswordReset email. This endpoint will be used by Chompy imports to send users "Activate Account" types of resets per [spec](https://docs.google.com/document/d/1RbWAAJA-zTfQYv6dBVIcpJqegKJUM0dH4DkVpikRpE0/edit?ts=5c6f0bd1#heading=h.f68k8ccmho5)

#### How should this be reviewed?

* Verify the Forgot Password webform (e.g. `/password/reset`) sends a Forgot Password email upon submitting a valid email

* Verify executing a `POST /resets` request with type `rock-the-vote-activate-account` sends an Activate Account email, prompting user to "Set Password"

#### Relevant Tickets

https://www.pivotaltracker.com/story/show/163779579

#### Checklist
- [x] Documentation added for changed endpoints.
